src/bsdfs/roughplastic.cpp

```diff
@@ -417,6 +417,12 @@ class RoughPlastic final : public BSDF<Float, Spectrum> {
         return oss.str();
     }
 
+    void set_grad_suspended(bool state) override {
+        if constexpr (ek::is_diff_array_v<Float>) {
+            ek::set_grad_suspended(m_external_transmittance, state);
```

Comment:
This won't ever be differntiable.

Reply:
I'm confused about what we should stop keeping track of and what we shouldn't. All differentiable parameters end up in `params` when using `traverse(scene)` don't they? I thought the point of this mechanism was to also stop recording the "hidden" parameters that are computed from differentiable parameters and that would then be recorded.

Reply:
Generally true, but in this specific case: external_transmittance is a precomputed array that's generated using a non-differentiable algorithm.